File: orden_por_columnas (1).cpp
#include <iostream>
#include <algorithm>
using namespace std;

/*
 * Tarea: Orden por columnas de matriz 2D (C++)
 * Cada columna se ordena de forma ascendente de manera independiente.
 */

const int FILAS = 3;
const int COLUMNAS = 3;

void imprimirMatriz(int matriz[FILAS][COLUMNAS]) {
    for (int fila = 0; fila < FILAS; fila++) {
        for (int col = 0; col < COLUMNAS; col++) {
            cout << matriz[fila][col] << "\t";
        }
        cout << endl;
    }
}

int main() {
    int matriz[FILAS][COLUMNAS] = {
        {8, 3, 5},
        {1, 9, 2},
        {7, 4, 6}
    };

    cout << "Matriz original:" << endl;
    imprimirMatriz(matriz);

    for (int col = 0; col < COLUMNAS; col++) {
        int columna[FILAS];
        for (int fila = 0; fila < FILAS; fila++) {
            columna[fila] = matriz[fila][col];
        }
        sort(columna, columna + FILAS);
        for (int fila = 0; fila < FILAS; fila++) {
            matriz[fila][col] = columna[fila];
        }
    }

    cout << "\nMatriz ordenada por columnas:" << endl;
    imprimirMatriz(matriz);

    return 0;
}

File: orden_por_columnas (1).py
"""
Tarea: Orden por columnas de matriz 2D (Python)
Cada columna se ordena de forma ascendente de manera independiente.
Ejecutar con: python orden_por_columnas.py
"""


def imprimir_matriz(matriz):
    for fila in matriz:
        print("\t".join(str(valor) for valor in fila))


matriz = [
    [8, 3, 5],
    [1, 9, 2],
    [7, 4, 6]
]

print("Matriz original:")
imprimir_matriz(matriz)

filas = len(matriz)
columnas = len(matriz[0])

for col in range(columnas):
    columna = [matriz[fila][col] for fila in range(filas)]
    columna.sort()
    for fila in range(filas):
        matriz[fila][col] = columna[fila]

print("\nMatriz ordenada por columnas:")
imprimir_matriz(matriz)
